### PR TITLE
Add Proxy support for Avatar

### PR DESCRIPTION
**Pull request by jchittoda**
- [x] Import OZ upgradability package and use `Initializable.sol`
- [x] Use GnosisSafeProxy contract for Avatar proxy
- [x] Delete NonUpgradableProxy
- [x] Deploy Proxy for an Avatar from Registry
- [x] Fix test engine
- [x] Use Proxy in tests

### CODE REVIEW

File: contracts/bprotocol/NonUpgradeableProxy.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
```
Comment (yaronvel) [inline contracts/bprotocol/NonUpgradeableProxy.sol:3]: this looks too complicated, would be best to use this:
https://github.com/gnosis/safe-contracts/blob/development/contracts/proxies/GnosisSafeProxy.sol

File: contracts/bprotocol/Registry.sol [hunk 2]
```diff
@@ -114,7 +114,9 @@ contract Registry is Ownable {
         require(avatarOf[_owner] == address(0), "Registry: avatar-exits-for-owner");
         // _owner should not be an avatar address
         require(ownerOf[_owner] == address(0), "Registry: cannot-create-an-avatar-of-avatar");
-        address _avatar = address(new Avatar(bComptroller, comptroller, comp, cEther, address(this)));
+        //TODO
+        // address _avatar = address(new Avatar(bComptroller, comptroller, comp, cEther, address(this)));
+        address _avatar = address(new Avatar());
```
Comment (yaronvel) [inline contracts/bprotocol/Registry.sol:119]: after this we should call initialize right?

File: contracts/bprotocol/avatar/Avatar.sol [hunk 3]
```diff
@@ -25,22 +27,15 @@ contract Avatar is AbsComptroller, AbsCToken {
      * @param _cETH cETH contract address
      * @param _registry Registry contract address
      */
-    constructor(
+    function initialize(
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/Avatar.sol:30]: let's have only registry as input, to save deployment gas.

File: contracts/bprotocol/avatar/AvatarFactory.sol [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    constructor(address _registry, address _avatarImpl) public {
+        registry = _registry;
+        avatarImpl = _avatarImpl;
+    }
+
+    function newAvatar() external returns (address) {
+        bytes memory data = ""; // TODO call initialize
+        NonUpgradeableProxy proxy = new NonUpgradeableProxy(avatarImpl, data);
+        return proxy;
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/AvatarFactory.sol:18]: maybe here we should call avatar.initialize

**Review by yaronvel — CHANGES_REQUESTED** [review]

**Review by yaronvel — CHANGES_REQUESTED** [review]
get cETH, comp, comptroller and other values from registery.

File: .gitmodules [hunk 5]
```diff
@@ -6,3 +6,6 @@
 [submodule "user-rating"]
 	path = user-rating
 	url = https://github.com/backstop-protocol/user-rating.git
+[submodule "openzeppelin-upgrades"]
```
Comment (yaronvel) [inline .gitmodules:9]: can we remove this now?

**Review by yaronvel — APPROVED** [review]

File: .gitmodules (hunk 5 above)
Comment (jchittoda) [reply .gitmodules:9]: We are using `Initializable` to allow initialization only once.

**Review by yaronvel — APPROVED** [review]

**Pull request merged**